### PR TITLE
Minor performance improvements.

### PR DESCRIPTION
**Pull request by k163377**
I am not good at English, so I apologize if I am rude.

## Description
This PR summarizes minor performance improvements.

## Checklist

- [x] PR description added
- [ ] tests are added
- [ ] `CHANGELOG.md` is updated

In case of adding a new rule:
- [ ] `README.md` is updated
- [ ] Rule has been applied on Ktlint itself and violations are fixed

### CODE REVIEW

File: ktlint-ruleset-standard/src/main/kotlin/com/pinterest/ktlint/ruleset/standard/ImportOrderingRule.kt [hunk 1]
```diff
@@ -147,6 +147,37 @@ public class ImportOrderingRule :
...
+        val importTextSet = mutableSetOf<String>()
+
+        children.forEach { current ->
+            val isPsiWhiteSpace = current.psi is PsiWhiteSpace
+
+            if (
+                current.elementType == ElementType.IMPORT_DIRECTIVE ||
+                isPsiWhiteSpace && current.textLength > 1 // also collect empty lines, that are represented as "\n\n"
+            ) {
+                if (isPsiWhiteSpace || importTextSet.add(current.text)) {
```
Comment (k163377) [inline ktlint-ruleset-standard/src/main/kotlin/com/pinterest/ktlint/ruleset/standard/ImportOrderingRule.kt:165]: These lines are the most significant improvements in this PR.
The existing code was inefficient at checking for duplicate `imports`, but this change reduces the amount of computation.
(However, this is generally only a minor improvement, as the use case of thousands of `import` lines in a single file is unlikely.)

File: ktlint-core/src/main/kotlin/com/pinterest/ktlint/core/internal/EditorConfigGenerator.kt [hunk 2]
```diff
@@ -39,25 +39,20 @@ internal class EditorConfigGenerator(
...
+                        "${prop.type.name} = $value"
                     }
-                }
-
-                acc
-            }
-            .filterValues { it != null }
-            .map {
-                "${it.key} = ${it.value}"
+                } ?: emptyList()
```
Comment (paul-dingemans) [inline ktlint-core/src/main/kotlin/com/pinterest/ktlint/core/internal/EditorConfigGenerator.kt:55]: When reviewing this code change, I spotted a bug in the existing code. If the same property is used by multiple rules, it is also printed multiple times in the output. I will fix that.

File: ktlint-ruleset-experimental/src/main/kotlin/com/pinterest/ktlint/ruleset/experimental/BlockCommentInitialStarAlignmentRule.kt [hunk 3]
```diff
@@ -31,7 +31,7 @@ class BlockCommentInitialStarAlignmentRule :
             val modifiedLines = mutableListOf<String>()
             lines.forEach { line ->
                 val modifiedLine =
-                    Regex("^([\t ]+\\*)(.*)$")
+                    secondAndLaterLineCommentRegex
```
Comment (paul-dingemans) [inline ktlint-ruleset-experimental/src/main/kotlin/com/pinterest/ktlint/ruleset/experimental/BlockCommentInitialStarAlignmentRule.kt:34]: I will rename this constant to `continuationCommentRegex`

File: ktlint-ruleset-standard/src/main/kotlin/com/pinterest/ktlint/ruleset/standard/ImportOrderingRule.kt [hunk 4]
```diff
@@ -147,6 +147,37 @@ public class ImportOrderingRule :
...
+    ): Pair<Boolean, List<ASTNode>> {
+        var autoCorrectDuplicateImports = false
+        val imports = mutableListOf<ASTNode>()
+        val importTextSet = mutableSetOf<String>()
+
+        children.forEach { current ->
+            val isPsiWhiteSpace = current.psi is PsiWhiteSpace
+
+            if (
+                current.elementType == ElementType.IMPORT_DIRECTIVE ||
```
Comment (paul-dingemans) [inline ktlint-ruleset-standard/src/main/kotlin/com/pinterest/ktlint/ruleset/standard/ImportOrderingRule.kt:162]: Lines 161 and 162 should be merged. Fixed by me.

File: ktlint-ruleset-standard/src/main/kotlin/com/pinterest/ktlint/ruleset/standard/ImportOrderingRule.kt (hunk 1 above)
Comment (paul-dingemans) [reply ktlint-ruleset-standard/src/main/kotlin/com/pinterest/ktlint/ruleset/standard/ImportOrderingRule.kt:165]: The performance improvement is neglible. I do like the improved readability.

**Comment by k163377** [comment]
Thanks for review.

**Pull request merged**